### PR TITLE
Replace ankr rpc url for any sepolia links

### PR DESCRIPTION
**Pull request by jlmonroy13**
### Task
https://linear.app/earthfast/issue/ENG-74/replace-ankr-rpc-because-it-sucks-move-to-alchemy-or-public-infura-in

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 dist
 node_modules
 oclif.manifest.json
+.earthfast/rpc_api_key
```
Comment (dmanjunath) [inline .gitignore:5]: this should go outside the repository in your computer home dir like ~/.earthfast/rpc_api_key. there's already a ~/.earthfast dir that we have

File: src/networks.ts [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
+let RPC_API_KEY: string | null = null;
+const keyPath = path.join(process.cwd(), ".earthfast", "rpc_api_key");
+
+if (fs.existsSync(keyPath)) {
+  RPC_API_KEY = fs.readFileSync(keyPath, "utf-8").trim();
+} else if (process.env.RPC_API_KEY) {
+  RPC_API_KEY = process.env.RPC_API_KEY;
+} else {
+  throw new Error(
+    "No RPC API key found. Please set it in .earthfast/rpc_api_key or as an environment variable RPC_API_KEY"
```
Comment (dmanjunath) [inline src/networks.ts:20]: i'm debating erroring out, i think maybe we should fallback to the ankr rpc so user can still continue the action. but lets print a message like "no alchemy API key provided falling back to ankr"

maybe you can make a rpcUrl variable above and default that to ankr and override with alchemy if api key is provided

File: src/networks.ts [hunk 3]
```diff
@@ -1,19 +1,35 @@
...
   url: string;
   abi?: string;
 }
 
+let RPC_API_KEY: string | null = null;
+const keyPath = path.join(process.cwd(), ".earthfast", "rpc_api_key");
+
+if (fs.existsSync(keyPath)) {
+  RPC_API_KEY = fs.readFileSync(keyPath, "utf-8").trim();
+} else if (process.env.RPC_API_KEY) {
```
Comment (dmanjunath) [inline src/networks.ts:16]: if you do npm run build and pass in the RPC_API_KEY as an environment variable does it have the correct value?

Comment (jlmonroy13) [reply src/networks.ts:16]: I have some doubts regarding this, can we chat about it on tomorrow meeting

Comment (dmanjunath) [reply src/networks.ts:16]: Yes for sure

File: src/networks.ts (hunk 2 above)
Comment (cheran-senthil) [reply src/networks.ts:20]: +1 to a fallback url

**Pull request closed without merging**